### PR TITLE
Fix part #44: Full UI profile pin/password screen.

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR fixes UI for profile pin/ password screen.

### Mock
https://xd.adobe.com/view/0687f00c-695b-437a-79a6-688e7f4f7552-70b6/screen/d59621be-cceb-4686-9442-68dd6b3e2ee2/PC-AP-Filled-

https://xd.adobe.com/view/0687f00c-695b-437a-79a6-688e7f4f7552-70b6/screen/dc5dd024-cff4-4419-80a5-24147371bb72/PC-NP-Empty-Error-

### Accessibility Scanner

- [x] Passes Accessibility scanner test

### Screenshot
Screenshot_20200109-152224
Screenshot_20200109-152233

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml:18]: Press `Option+Cmd+L` to organise everything.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Change this color to `oppiaPrimaryText`

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Rename this to `hello_text_view` to keep it consistent with other views too.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Rename this to `enter_text_view` to keep it consistent with other views too.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Start and end margin is 60dp in this.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Ditto, change text color

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Remove this linespacing, I think android takes care of it on its own, and also, we have not used anywhere in the entire app.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Check this mock, 
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/dc5dd024-cff4-4419-80a5-24147371bb72/PC-NP-Empty-Error-

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml:70]: itemRadius is `2dp` not `4dp`
lineColor is also different in mocks, `#00000029` this code is from mocks, you will need to convert this for android.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: This should be 4 dp, because you already have a 8dp padding in pinView.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Remove this lineSpacing

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml:100]: primaryColor in the application is `#00645C` and the color in the mocks is `#009688` , but I think it was because of the fact that we had updated the mocks color in between. Maybe trying to clear this with Chantel, which color do we need here.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: We cannot have a negative margin.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: We cannot have a negative margin.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml:127]: The color code of this text view in mocks in `#008098`

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: This color is also different, check mocks. Replace this with `oppiaLightGreen`

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml:18]: Done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Updated.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Updated.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Updated.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml:127]: Changed textcolor to #008098.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml:100]: @mschanteltc can you please clear, which color we are supposed to keep for the text `I forgot pin`

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Removed.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Removed.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Updated.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Thanks for this mock @rt4914 . Updated accordingly.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml:70]: I tried this color. It looks too light.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (nikitamarysolomanpvt) [inline app/src/main/res/layout/pin_password_activity.xml]: nit change please make the commonly used widget naming with file name to avoid renaming issues.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [reply app/src/main/res/layout/pin_password_activity.xml:70]: Color code should be `#4A000000`

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [reply app/src/main/res/layout/pin_password_activity.xml]: This has not been updated yet.

**Review by rt4914 — APPROVED** [review]
LGTM, marked some conversations as unresolved.

**Review by nikitamarysolomanpvt — CHANGES_REQUESTED** [review]
Please confirm regarding the border with @mschanteltc  because in mock there are not borders to the pinview. Also please use proper eye icons as per mock.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml:70]: done.

**Comment by veena14cs** [comment]
> Please confirm regarding the border with @mschanteltc because in mock there are not borders to the pinview. Also please use proper eye icons as per mock.

@mschanteltc can you please provide me icons to show and hide password.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (mschanteltc) [reply app/src/main/res/layout/pin_password_activity.xml:100]: It would be #00645C, but against a #F0FFFF background, it would pass WCAG AA but not WCAG AAA. It is 0.14 shy off from the minimum requirement, but it is possible we can make this an exception?

**Comment by mschanteltc** [comment]
> > Please confirm regarding the border with @mschanteltc because in mock there are not borders to the pinview. Also please use proper eye icons as per mock.
> 
> @mschanteltc can you please provide me icons to show and hide password.

Yes, it's available in the Android Image drive under the folder "Profile Chooser."

**Comment by mschanteltc** [comment]
Also there is no border, but there is a #000000 shadow at 30% opacity with a blur of 3.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: > > > Please confirm regarding the border with @mschanteltc because in mock there are not borders to the pinview. Also please use proper eye icons as per mock.
> > 
> > 
> > @mschanteltc can you please provide me icons to show and hide password.
> 
> Yes, it's available in the Android Image drive under the folder "Profile Chooser."

There is slight confusion in the show/hide icon along with the text. As per the [mock](https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/686c27d1-f07b-4680-95da-7767d71558a4/PC-AP-Empty-/) hide_eye_icon is been used with show text. But in our implementation it is other way. Which one is correct in this.
Screenshot_20200110-111234
Screenshot_20200110-111244

**Comment by veena14cs** [comment]
> > > Please confirm regarding the border with @mschanteltc because in mock there are not borders to the pinview. Also please use proper eye icons as per mock.
> > 
> > 
> > @mschanteltc can you please provide me icons to show and hide password.
> 
> Yes, it's available in the Android Image drive under the folder "Profile Chooser."

There is slight confusion in the show/hide icon along with the text. As per the [mock](https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/686c27d1-f07b-4680-95da-7767d71558a4/PC-AP-Empty-/) hide_eye_icon is been used with show text. But in our implementation it is other way. Which one is correct in this.
Screenshot_20200110-111234
Screenshot_20200110-111244

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml:100]: Thanks @mschanteltc for clarification I will be using #00645C for textcolor against a #F0FFFF background.

**Comment by mschanteltc** [comment]
> > > > Please confirm regarding the border with @mschanteltc because in mock there are not borders to the pinview. Also please use proper eye icons as per mock.
> > > 
> > > 
> > > @mschanteltc can you please provide me icons to show and hide password.
> > 
> > 
> > Yes, it's available in the Android Image drive under the folder "Profile Chooser."
> 
> There is slight confusion in the show/hide icon along with the text. As per the [mock](https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/686c27d1-f07b-4680-95da-7767d71558a4/PC-AP-Empty-/) hide_eye_icon is been used with show text. But in our implementation it is other way. Which one is correct in this.
> Screenshot_20200110-111234
> Screenshot_20200110-111244

I agree with your implementation. I updated the mocks to reflect this.

**Comment by veena14cs** [comment]
> > > > > Please confirm regarding the border with @mschanteltc because in mock there are not borders to the pinview. Also please use proper eye icons as per mock.
> > > > 
> > > > 
> > > > @mschanteltc can you please provide me icons to show and hide password.
> > > 
> > > 
> > > Yes, it's available in the Android Image drive under the folder "Profile Chooser."
> > 
> > 
> > There is slight confusion in the show/hide icon along with the text. As per the [mock](https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/686c27d1-f07b-4680-95da-7767d71558a4/PC-AP-Empty-/) hide_eye_icon is been used with show text. But in our implementation it is other way. Which one is correct in this.
> > Screenshot_20200110-111234
> > Screenshot_20200110-111244
> 
> I agree with your implementation. I updated the mocks to reflect this.

Thanks for clarification.

**Comment by veena14cs** [comment]
I have added Shadow effect to pinview. PTAL once again though it is approved.

Screenshot_20200113-114529

**Review by rt4914 — CHANGES_REQUESTED** [review]
The eye icon and the status textview just below that have different colors. I think both of these should be `#008098`

**Comment by veena14cs** [comment]
> 008098

I have updated icon color. 
There is no consistent color for `I forgot my pin` and `show/hide` icons and status. @mschanteltc can you please clarify this, whether we need to keep it that way or change it.

**Comment by veena14cs** [comment]
I have updated shadow with lighter shades as it was not matching with mock. @mschanteltc PTAL 
Screenshot_20200113-121942

**Comment by nikitamarysolomanpvt** [comment]
> I have updated shadow with lighter shades as it was not matching with mock. @mschanteltc PTAL
> Screenshot_20200113-121942

this drop shadow looks better but please update color code to match mock

**Comment by veena14cs** [comment]
> > I have updated shadow with lighter shades as it was not matching with mock. @mschanteltc PTAL
> > Screenshot_20200113-121942
> 
> this drop shadow looks better but please update color code to match mock

PTAL now. 
Screenshot_20200113-125334

**Comment by mschanteltc** [comment]
> > 008098
> 
> I have updated icon color.
> There is no consistent color for `I forgot my pin` and `show/hide` icons and status. @mschanteltc can you please clarify this, whether we need to keep it that way or change it.

Sorry about the inconsistency. Let's make the icon and the text be #00645C.

**Comment by mschanteltc** [comment]
> > > I have updated shadow with lighter shades as it was not matching with mock. @mschanteltc PTAL
> > > Screenshot_20200113-121942
> > 
> > 
> > this drop shadow looks better but please update color code to match mock
> 
> PTAL now.
> Screenshot_20200113-125334

Now looking at the shadows, it's difficult to distinguish where the outlines of the boxes are. Can we remove the shadows and add a #707070 border around them, width of 0.5 px if possible? I updated the 7 DRAFT mock to reflect this.

**Comment by veena14cs** [comment]
> 00645C

Okay Updated icon and textcolor to 00645C

**Comment by veena14cs** [comment]
> > 00645C
> 
Okay, updated icon and textcolor to 00645C

**Comment by veena14cs** [comment]
Done changes as per the updated mocks in [v7_FINALIZED](https://xd.adobe.com/view/0687f00c-695b-437a-79a6-688e7f4f7552-70b6/screen/d59621be-cceb-4686-9442-68dd6b3e2ee2/PC-AP-Filled-)

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Remove this unused attribute

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml:117]: Add contentDescription to this image.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Change this to `android:paddingTop="12dp"` for better clickable item

File: app/src/main/res/values/colors.xml
Comment (rt4914) [inline app/src/main/res/values/colors.xml]: Remove this color. It is not getting used anywhere.

File: app/src/main/res/values/strings.xml
Comment (rt4914) [inline app/src/main/res/values/strings.xml]: I am not sure if this `\n` should be used as it means that even on landscape mode it will contain 2 lines. Can you just confirm with @mschanteltc regarding this.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Generally good. Left some nit comments.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml:117]: Done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Done.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: updated.

File: app/src/main/res/values/strings.xml
Comment (veena14cs) [reply app/src/main/res/values/strings.xml]: @mschanteltc do we have this text `Please enter your Administrator PIN` in 2 lines in landscape mock or its a single line text in landscape. Please confirm.

File: app/src/main/res/values/colors.xml
Comment (veena14cs) [reply app/src/main/res/values/colors.xml]: Removed.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM

File: app/src/main/res/values/strings.xml
Comment (mschanteltc) [reply app/src/main/res/values/strings.xml]: In landscape mode, the text will be one line. So the text in portrait mode should be bounded by a text box size. My mocks have it so that is is 240px wide.

Comment (veena14cs) [reply app/src/main/res/values/strings.xml]: Thanks @mschanteltc made textbox size as 240px wide.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (rt4914) [inline app/src/main/res/layout/pin_password_activity.xml]: Remove this, layout_height should be `wrap_content` and use `minHeight` - `48dp`

File: app/src/main/res/values/strings.xml
Comment (veena14cs) [reply app/src/main/res/values/strings.xml]: To avoid giving fixed width. Created dimens file for different screen size, to have double line text in portrait mode and single line in landscape.

File: app/src/main/res/values-xhdpi/dimens.xml
Comment (rt4914) [inline app/src/main/res/values-xhdpi/dimens.xml]: Remove these other dimens from this file. Just keep only those in which there is difference in value

**Review by rt4914 — COMMENTED** [review]
One change in dimens file, else looks great.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (veena14cs) [reply app/src/main/res/layout/pin_password_activity.xml]: Updated.

File: app/src/main/res/values-xhdpi/dimens.xml
Comment (veena14cs) [reply app/src/main/res/values-xhdpi/dimens.xml]: Done.

**Review by rt4914 — APPROVED** [review]
LGTM.

**Comment by rt4914** [comment]
@veena14cs Lets not block this PR anymore just because of two different `dimens.xml` file. I will keep track of this and discuss this with @BenHenning and meanwhile you can merge this PR.

**Comment by veena14cs** [comment]
> @veena14cs Lets not block this PR anymore just because of two different `dimens.xml` file. I will keep track of this and discuss this with @BenHenning and meanwhile you can merge this PR.

Okay will merge this PR thanks.

**Comment by BenHenning** [comment]
Having multiple dimension files seems sensible.

**Comment by veena14cs** [comment]
> Having multiple dimension files seems sensible.

Got it thanks.

**Pull request merged**
